File: Data Structures & Algorithms/eating-bananas/submission-1.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int maxN = 0;
        for(int p: piles){
            maxN = max(maxN, p);
        }
        int l = 1, r = maxN;
        int res=maxN;
        while(l <= r){
            int mid = (r+l) / 2;
            int total = 0;
            for(int p: piles){
                double dp = p;
                int n = ceil(dp/mid);
                total += n;
            }
            if(total <= h){
                res = mid;
                r = mid - 1;
            }else{
                l = mid + 1;
            }

        }
        return res;
    }
};

File: Data Structures & Algorithms/partition-labels/submission-1.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        dp = {}
        for i, c in enumerate(s):
            if i >= dp.get(c, 0):
                dp[c] = i
        l, r = 0, dp[s[0]]
        res = []
        for i, c in enumerate(s):
            r = max(dp[c], r)
            if i == r:
                res.append(r - l + 1)
                l = i + 1
            
        return res